presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/PartitionTable.java

```diff
@@ -0,0 +1,455 @@
+import io.prestosql.spi.type.TimeZoneKey;
+import io.prestosql.spi.type.TypeManager;
+import io.prestosql.spi.type.TypeUtils;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.PartitionField;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.types.Comparators;
+import org.apache.iceberg.types.Conversions;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.StructLikeWrapper;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.prestosql.plugin.iceberg.IcebergUtil.getIdentityPartitions;
+import static io.prestosql.plugin.iceberg.TypeConveter.toPrestoType;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.DateTimeEncoding.packDateTimeWithZone;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toCollection;
+import static java.util.stream.Collectors.toSet;
+
+public class PartitionTable
+        implements SystemTable
+{
+    private final IcebergTableHandle tableHandle;
+    private final ConnectorSession session;
+    private final TypeManager typeManager;
+    private final Table icebergTable;
+    private Map<Integer, Type.PrimitiveType> idToTypeMapping;
+    private List<Types.NestedField> nonPartitionPrimitiveColumns;
+    private List<io.prestosql.spi.type.Type> partitionColumnTypes;
+    private List<io.prestosql.spi.type.Type> resultTypes;
+    private List<RowType> columnMetricTypes;
+
+    public PartitionTable(IcebergTableHandle tableHandle, ConnectorSession session, TypeManager typeManager, Table icebergTable)
+    {
+        this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
+        this.session = requireNonNull(session, "session is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        this.idToTypeMapping = icebergTable.schema().columns().stream()
```

Comment:
It is necessary to change the current thread context class loader to the iceberg plugin class loader.